src/Ionide.ProjInfo/Library.fs

```diff
@@ -392,9 +385,88 @@ module ProjectLoader =
                 with set (v: LoggerVerbosity): unit = ()
         }
 
-    let getTfm (path: string) readingProps isLegacyFrameworkProj =
-        let pi = ProjectInstance(path, globalProperties = readingProps, toolsVersion = null)
+    let mergeGlobalProperties (collection: ProjectCollection) (otherProperties: IDictionary<string, string>) =
+        let combined = Dictionary(collection.GlobalProperties)
+
+        for kvp in otherProperties do
+            combined.Add(kvp.Key, kvp.Value)
+
+        combined
+
+    // it's _super_ important that the 'same' project (path + properties) is only created once in a project collection, so we have to check on this here
+    let findOrCreateMatchingProject path (collection: ProjectCollection) globalProps =
+        let createNewProject properties =
+            try
+                Project(
+                    projectFile = path,
+                    projectCollection = collection,
+                    globalProperties = properties,
+                    toolsVersion = null,
+                    loadSettings =
+                        (ProjectLoadSettings.IgnoreMissingImports
+                         ||| ProjectLoadSettings.IgnoreInvalidImports)
+                )
+            with :? System.InvalidOperationException as ex ->
+
+                // if the project is already loaded throw a nicer message
+                let message = System.Text.StringBuilder()
+
+                message
+                    .AppendLine("The project '{path}' already exists in the project collection with the same global properties.")
+                    .AppendLine("The global properties requested were:")
+                |> ignore
+
+                for (KeyValue(k, v)) in properties do
+                    message.AppendLine($"  {k} = {v}")
+                    |> ignore
+
+                message.AppendLine()
+                |> ignore
+
+                message.AppendLine("There are projects of the following properties already in the collection:")
+                |> ignore
+
+                for project in collection.GetLoadedProjects(path) do
+                    message.AppendLine($"Evaluation #{project.LastEvaluationId}")
+                    |> ignore
+
+                    for (KeyValue(k, v)) in project.GlobalProperties do
+                        message.AppendLine($"  {k} = {v}")
+                        |> ignore
+
+                    message.AppendLine()
+                    |> ignore
+
+                failwith (message.ToString())
```

Comment:
It'll be worth creating our own exception and setting the inner exception to the one we caught so we get that stacktrace.

Reply:
done!